File: LeetCode/376摆动序列.java
class Solution {
    public int wiggleMaxLength(int[] nums) {
        if (nums.length <= 1) return nums.length;
        int pre = 0, cur = 0, ans = 1;
        for (int i = 0; i < nums.length - 1; i++) {
            cur = nums[i + 1] - nums[i];
            if ((cur > 0 && pre <= 0) || (pre >= 0 && cur < 0)) {
                ans++;
                pre = cur;
            }
        }
        return ans;
    }
}

File: LeetCode/376摆动序列.go
func wiggleMaxLength(nums []int) (ans int) {
	ans = 1
	pre, cur := 0, 0
	lens := len(nums)
	if lens <= 1 {
		return lens
	}
	for i := 0; i < lens-1; i++ {
		cur = nums[i+1] - nums[i]
		if (pre <= 0 && cur > 0) || (pre >= 0 && cur < 0) {
			pre = cur
			ans++
		}
	}
	return ans
}